x-pack/plugins/index_management/server/lib/fetch_indices.ts

```diff
@@ -5,99 +5,57 @@
-  settings: {
-    index: {
-      hidden: 'true' | 'false';
-    };
-  };
-}
-
-interface GetIndicesResponse {
-  [indexName: string]: IndexInfo;
-}
-
 async function fetchIndicesCall(
-  callAsCurrentUser: CallAsCurrentUser,
+  client: IScopedClusterClient,
   indexNames?: string[]
 ): Promise<Index[]> {
   const indexNamesString = indexNames && indexNames.length ? indexNames.join(',') : '*';
 
   // This call retrieves alias and settings (incl. hidden status) information about indices
-  const indices: GetIndicesResponse = await callAsCurrentUser('transport.request', {
-    method: 'GET',
-    // transport.request doesn't do any URI encoding, unlike other JS client APIs. This enables
-    // working with Logstash indices with names like %{[@metadata][beat]}-%{[@metadata][version]}.
-    path: `/${encodeURIComponent(indexNamesString)}`,
-    query: {
-      expand_wildcards: 'hidden,all',
-    },
+  const { body: indices } = await client.asCurrentUser.indices.get({
+    index: indexNamesString,
+    expand_wildcards: 'hidden,all',
   });
 
   if (!Object.keys(indices).length) {
     return [];
   }
 
-  const catQuery: Pick<CatIndicesParams, 'format' | 'h'> & {
-    expand_wildcards: string;
-    index?: string;
-  } = {
+  const { body: catHits } = await client.asCurrentUser.cat.indices({
     format: 'json',
     h: 'health,status,index,uuid,pri,rep,docs.count,sth,store.size',
     expand_wildcards: 'hidden,all',
     index: indexNamesString,
-  };
-
-  // This call retrieves health and other high-level information about indices.
-  const catHits: Hit[] = await callAsCurrentUser('transport.request', {
-    method: 'GET',
-    path: '/_cat/indices',
-    query: catQuery,
   });
 
   // System indices may show up in _cat APIs, as these APIs are primarily used for troubleshooting
   // For now, we filter them out and only return index information for the indices we have
   // In the future, we should migrate away from using cat APIs (https://github.com/elastic/kibana/issues/57286)
   return catHits.reduce((decoratedIndices, hit) => {
-    const index = indices[hit.index];
+    const index = indices[hit.index!];
 
     if (typeof index !== 'undefined') {
-      const aliases = Object.keys(index.aliases);
+      const aliases = Object.keys(index.aliases!);
 
       decoratedIndices.push({
-        health: hit.health,
-        status: hit.status,
-        name: hit.index,
-        uuid: hit.uuid,
-        primary: hit.pri,
-        replica: hit.rep,
+        health: hit.health!,
+        status: hit.status!,
+        name: hit.index!,
+        uuid: hit.uuid!,
+        primary: hit.pri!,
+        replica: hit.rep!,
         documents: hit['docs.count'],
         size: hit['store.size'],
         isFrozen: hit.sth === 'true', // sth value coming back as a string from ES
         aliases: aliases.length ? aliases : 'none',
+        // @ts-expect-error Property 'index' does not exist on type 'IndicesIndexSettings | IndicesIndexStatePrefixedSettings'.
```

Comment:
I believe any ts errors due to lack of coverage/incorrect types in the es js client should be commented as ```// @ts-expect-error @elastic/elasticsearch ....```

Reply:
I'm not sure it's incorrect typing - according to `@elastic/elasticsearch` typings it should be refactored to `hidden: index.settings.hidden`

Reply:
I just tested this and verified that the original path (`index.settings.index.hidden`) is correct. I've updated all the annotations to be `// @ts-expect-error @elastic/elasticsearch`.

Reply:
@cjcenizal if you are sure there is a problem you should report it to the client team.

Reply:
Thanks. This is listed as a follow-up item in the PR description.